aspnet/publishing/linuxproduction.rst

```diff
@@ -0,0 +1,190 @@
+
+
+Installing supervisor
+~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+    sudo apt-get install supervisor
+
+.. note::
+
+    ``supervisor`` is a python based tool and you can acquire it through `pip <http://supervisord.org/installing.html#installing-via-pip>`_ or `easy_install <http://supervisord.org/installing.html#internet-installing-with-setuptools>`_ instead.
+
+
+Configuring supervisor
+~~~~~~~~~~~~~~~~~~~~~~
+
+Supervisor works by creating child processes based on data in its configuration file. When a child process dies, supervisor is notified via the ``SIGCHILD`` signal and supervisor can react accordingly and restart your web application.
+
+To have supervisor monitor our application, we will add a file to the ``/etc/supervisor/conf.d/`` directory.
+
+.. code-block:: ini
+    :caption: /etc/supervisor/conf.d/hellomvc.conf
+
+    [program:hellomvc]
+    command=bash /var/aspnet/HelloMVC/approot/web
+    autostart=true
+    autorestart=true
+    stderr_logfile=/var/log/hellomvc.err.log
+    stdout_logfile=/var/log/hellomvc.out.log
+    environment=ASPNET_ENV=Production
+    user=aspnet
+    stopsignal=INT
+
+Once you are done editing the configuration file, restart the ``supervisord`` process to change the set of programs controlled by supervisord.
+
+.. code-block:: bash
+
+    #!/bin/sh
+    sudo service supervisor stop
+    sudo service supervisor start
+
+Start our web application on startup
+------------------------------------
+
+In our case, since we are using supervisor to manage our application, the application will be automatically started by supervisor. Supervisor uses a System V Init script to run as a daemon on system boot and will susbsequently launch your application. If you chose not to use supervisor or an equivalent tool, you will need to write a ``systemd`` or ``upstart`` or ``SysVinit`` script to start your application on startup.
+
+Recovering from an ungraceful shutdown
+--------------------------------------
+
+If your web application is terminated with a ``SIGKILL`` signal or the if host experiences a loss of power, ``Kestrel`` will not shut down gracefully and remove the socket file. To prevent subsequents attempts to restart your application from failing due to ``EADDRINUSE address already in use``, you can modify the shell script used to bootstrap your application to remove the socket file if present.
+
+.. code-block:: bash
+    :caption: /var/aspnet/HelloMVC/approot/web
+
+    if [ -f "/var/aspnet/HelloMVC/kestrel.sock" ]; then
+      rm "/var/aspnet/HelloMVC/kestrel.sock"
+    fi
+
+
+Viewing logs
+------------
+
+**Supervisord** logs messages about its own health and its subprocess' state changes to the activity log. The path to the activity log is configured via the ``logfile`` parameter in the configuration file.
+
+.. code-block:: bash
+
+    tail -f /var/log/supervisor/supervisord.log
```

Comment:
I think there's a `supervisorctl` command for it too.

I think this should use `sudo`.